src/orion/algo/evolution_es.py

```diff
@@ -0,0 +1,263 @@
+
+        self.hurdles = []
+
+        self.population = {}
+        for key in range(len(self.space)):
+            if not key == self.fidelity_index:
+                self.population[key] = -1 * np.ones(nums_population)
+
+        self.performance = np.inf * np.ones(nums_population)
+
+        self.budgets = compute_budgets(self.min_resources, self.max_resources,
+                                       self.reduction_factor, nums_population, pair)
+
+        self.brackets = [
+            BracketEVES(self, bracket_budgets, 1, space)
+            for bracket_budgets in self.budgets
+        ]
+        self.seed_rng(seed)
+
+    def _get_bracket(self, point):
+        """Get the bracket of a point during observe"""
+        return self.brackets[0]
+
+
+class BracketEVES(Bracket):
+    """Bracket of rungs for the algorithm Hyperband.
+
+    Parameters
+    ----------
+    evolutiones: `evolutiones` algorithm
+        The evolutiones algorithm object which this bracket will be part of.
+    budgets: list of tuple
+        Each tuple gives the (n_trials, resource_budget) for the respective rung.
+    repetition_id: int
+        The id of hyperband execution this bracket belongs to
+
+    """
+
+    def __init__(self, evolution_es, budgets, repetition_id, space):
+        super(BracketEVES, self).__init__(evolution_es, budgets, repetition_id)
+        self.eves = self.hyperband
+        self.space = space
+        self.search_space_remove_fidelity = []
+
+        if evolution_es.mutate_attr:
+            self.mutate_attr = copy.deepcopy(evolution_es.mutate_attr)
+        else:
+            self.mutate_attr = {}
+
+        function_string = self.mutate_attr.pop('function',
+                                               "orion.algo.mutate_functions.default_mutate")
+        mod_name, func_name = function_string.rsplit('.', 1)
+        mod = importlib.import_module(mod_name)
+        self.mutate_func = getattr(mod, func_name)
+
+        for i in range(len(space.values())):
+            if not i == self.eves.fidelity_index:
+                self.search_space_remove_fidelity.append(i)
+
+    def get_candidates(self, rung_id):
+        """Get a candidate for promotion"""
+        if self.has_rung_filled(rung_id + 1):
+            return []
+
+        rung = self.rungs[rung_id]['results']
+
+        population_range = (self.eves.nums_population
+                            if len(list(rung.values())) > self.eves.nums_population
+                            else len(list(rung.values())))
+
+        for i in range(population_range):
+            for j in self.search_space_remove_fidelity:
+                self.eves.population[j][i] = list(rung.values())[i][1][j]
+            self.eves.performance[i] = list(rung.values())[i][0]
+
+        population_index = list(range(self.eves.nums_population))
+        red_team = np.random.choice(population_index, self.eves.nums_comp_pairs, replace=False)
```

Comment:
All sampling should be seeded to ensure reproducibility. The RNG of self.eves could be used for this.

Reply:
Fixed.